### PR TITLE
fix: canvas data in iframe wasn't applied in the fast-forward mode

### PR DESCRIPTION
**Pull request by Mark-Fenng**
In #853, I implemented the virtual dom for the fast-forward mode.  But there is a corner case I didn't deal with.
If a canvas element is created with image data (serialized to `rr_dataURL`) in an iframe, it will be rebuilt as an empty canvas without the initial image data, in the fast-forward mode (using Virtual Dom optimization).
A simple test case is also added in this PR to reproduce this corner case.

### CODE REVIEW

File: packages/rrdom/src/diff.ts [hunk 1]
```diff
@@ -136,14 +136,27 @@ export function diff(
...
+            // This canvas element is created with initial data in an iframe element. https://github.com/rrweb-io/rrweb/pull/944
+            if (rrCanvasElement.rr_dataURL !== null) {
+              const image = document.createElement('img');
+              image.src = rrCanvasElement.rr_dataURL;
+              image.onload = () => {
+                const ctx = (oldElement as HTMLCanvasElement).getContext('2d');
+                if (ctx) {
+                  ctx.drawImage(image, 0, 0, image.width, image.height);
+                }
+              };
```
Comment (Juice10) [inline packages/rrdom/src/diff.ts:150]: These should be flipped to insure onload isn't called before the event handler gets added
```suggestion
              image.onload = () => {
                const ctx = (oldElement as HTMLCanvasElement).getContext('2d');
                if (ctx) {
                  ctx.drawImage(image, 0, 0, image.width, image.height);
                }
              };
              image.src = rrCanvasElement.rr_dataURL;
```

**Review by Juice10 — CHANGES_REQUESTED** [review]
Good catch! Found one little tweak to make the code slightly more robust

File: packages/rrdom/src/diff.ts (hunk 1 above)
Comment (Mark-Fenng) [reply packages/rrdom/src/diff.ts]: Flipped one is more straightforward for sure. But I suddenly realize that according to the JS engine's execution mechanism, I think the onload callback won't be called before the function definition. 
image.onload = ... definition is synchronous while the onload callback is asynchronous so that the callback will be always invoked after the function definition.

**Review by Yuyz0112 — APPROVED** [review]
